corehq/apps/reports/util.py

```diff
@@ -116,15 +116,17 @@ def _create_temp_user(user_id):
         if not user_filter:
             user_filter = HQUserType.all()
         users = []
-        submitted_user_ids = get_all_user_ids_submitted(domain)
-        registered_user_ids = dict([(user.user_id, user) for user in CommCareUser.by_domain(domain)])
+        submitted_user_ids = set(get_all_user_ids_submitted(domain))
+        registered_users_by_id = dict([(user.user_id, user) for user in CommCareUser.by_domain(domain)])
         if include_inactive:
-            registered_user_ids.update(dict([(u.user_id, u) for u in CommCareUser.by_domain(domain, is_active=False)]))
+            registered_users_by_id.update(dict(
+                [(u.user_id, u) for u in CommCareUser.by_domain(domain, is_active=False)]
+            ))
```

Comment:
nit (fine to punt): 
```python
# this assembles a list, then converts that to a dict:
dict([(u._id, u) for u in users])
# this uses a generator so you never build that list:
dict((u._id, u) for u in users)
# this is my preference overall:
{u._id: u for u in users}
```
My hunch is that the performance differences are marginal compared to the db calls here, but it's still good to think about.